plugin-unmanaged-racks/config/config.go

```diff
@@ -0,0 +1,164 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+
+	"github.com/ODIM-Project/ODIM/plugin-unmanaged-racks/logging"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/satori/go.uuid"
+	"gopkg.in/yaml.v3"
+)
+
+type PluginConfig struct {
+	Host               string          `yaml:"Host" envconfig:"HOST"`
+	Port               string          `yaml:"Port" envconfig:"PORT"`
+	UserName           string          `yaml:"UserName" envconfig:"BASIC_AUTH_USERNAME"`
+	Password           string          `yaml:"Password" envconfig:"BASIC_AUTH_PASSWORD"`
+	RootServiceUUID    string          `yaml:"RootServiceUUID" envconfig:"SERVICE_ROOT_UUID"`
+	OdimNBUrl          string          `yaml:"OdimNBUrl" envconfig:"ODIM_NORTBOUNND_URL"`
+	FirmwareVersion    string          `yaml:"FirmwareVersion" envconfig:"FIRMWARE_VERSION"`
+	URLTranslation     *URLTranslation `yaml:"URLTranslation"`
+	TLSConf            *TLSConf        `yaml:"TLSConf"`
+	PKIRootCAPath      string          `yaml:"PKIRootCACertificatePath" envconfig:"PKI_ROOT_CA_PATH"`
+	PKIPrivateKeyPath  string          `yaml:"PKIPrivateKeyPath" envconfig:"PKI_PRIVATE_KEY_PATH"`
+	PKICertificatePath string          `yaml:"PKICertificatePath" envconfig:"PKI_CERTIFICATE_PATH_PATH"`
+	LogLevel           string          `yaml:"LogLevel" envconfig:"LOG_LEVEL"`
+	RedisAddress       string          `yaml:"RedisAddress" envconfig:"REDIS_ADDRESS"`
+	SentinelMasterName string          `yaml:"SentinelMasterName" envconfig:"SENTINEL_MASTER_NAME"`
+}
+
+type URLTranslation struct {
+	NorthBoundURL map[string]string `yaml:"NorthBoundURL"` // holds value of NorthBound Translation
+	SouthBoundURL map[string]string `yaml:"SouthBoundURL"` // holds value of SouthBound Translation
+}
+
+type TLSConf struct {
+	MinVersion            uint16   `yaml:"MinVersion"`
+	MaxVersion            uint16   `yaml:"MaxVersion"`
+	PreferredCipherSuites []uint16 `yaml:"PreferredCipherSuites"`
+}
+
+func ReadPluginConfiguration() (*PluginConfig, error) {
+	pc := new(PluginConfig)
+
+	if cp := os.Getenv("PLUGIN_CONFIG_FILE_PATH"); cp != "" {
+		if configData, err := ioutil.ReadFile(cp); err == nil {
+			_ = yaml.Unmarshal(configData, pc)
+		} else {
+			logging.Warnf("Cannot load configuration file: %s", err)
+		}
+	}
+
+	if err := envconfig.Process("PLUGIN", pc); err != nil {
+		logging.Warnf("Cannot load ENV configuration: %s", err)
+	}
+
+	return pc, validate(pc)
+}
+
+func validate(pc *PluginConfig) error {
+	if pc.LogLevel == "" {
+		pc.LogLevel = "debug"
+	}
+	if pc.OdimNBUrl == "" {
+		return fmt.Errorf("OdimraNBUrl has to be specified")
+	}
+
+	if _, e := url.Parse(pc.OdimNBUrl); e != nil {
+		return fmt.Errorf("given OdimraNBUrl is not correct URL")
+	}
+
+	if _, err := uuid.FromString(pc.RootServiceUUID); err != nil {
+		return err
+	}
+
+	if pc.FirmwareVersion == "" {
+		pc.FirmwareVersion = "1.0"
```

Comment:
in odim config connection method data is defined for urp  is having firmware version of `v1.0.0` please assign same as default value

Reply:
done, defaulting firmware version has been removed at all.